### PR TITLE
new machines marked as NOT WORKING - Aquaplus

### PR DESCRIPTION
**Pull request by mamehaze**
new machines marked as NOT WORKING
----------
Aquaplus P/ECE [David Haywood]

### CODE REVIEW

File: src/devices/cpu/s1c33/s1c33dasm.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	offs_t handle_101x_xx00_xxxx_xxxx(std::ostream &stream, offs_t pc, const s1c33_disassembler::data_buffer &opcodes, const s1c33_disassembler::data_buffer &params, u16 op);
+	offs_t handle_101x_xx01_xxxx_xxxx(std::ostream &stream, offs_t pc, const s1c33_disassembler::data_buffer &opcodes, const s1c33_disassembler::data_buffer &params, u16 op);
+	offs_t handle_101x_xx10_xxxx_xxxx(std::ostream &stream, offs_t pc, const s1c33_disassembler::data_buffer &opcodes, const s1c33_disassembler::data_buffer &params, u16 op);
+	offs_t handle_101x_xx11_xxxx_xxxx(std::ostream &stream, offs_t pc, const s1c33_disassembler::data_buffer &opcodes, const s1c33_disassembler::data_buffer &params, u16 op);
+	offs_t handle_110x_xxxx_xxxx_xxxx(std::ostream &stream, offs_t pc, const s1c33_disassembler::data_buffer &opcodes, const s1c33_disassembler::data_buffer &params, u16 op);
+	offs_t handle_111x_xxxx_xxxx_xxxx(std::ostream &stream, offs_t pc, const s1c33_disassembler::data_buffer &opcodes, const s1c33_disassembler::data_buffer &params, u16 op);
```
Comment (cuavas) [inline src/devices/cpu/s1c33/s1c33dasm.h:45]: You don’t need `s1c33_disassembler::` inside the `s1c33_disassembler` class declaration when there are no base classes that depend on template arguments (the class isn’t even a template).

**Review by cuavas — CHANGES_REQUESTED** [review]
Please make the disassemble use the official assembler syntax shown in the architecture manual.

Also, please use the `0x` prefix for hex literals in the disassembly, as used in the examples in the architecture manual (the assembler defaults to decimal).

**Comment by mamehaze** [comment]
I'd rather use syntax I'm comfortable with in my head for now, while I develop this thing.

It can be changed later if needs be, unless you're nuking this progress before it can really get started too?

**Comment by cuavas** [comment]
The standard in MAME is to use syntax as close to the official assembler as reasonably possible.  It isn’t going to be merged in this state.

If this is some intermediate state while you’re developing it, please close the pull request and develop it on a branch yourself until it’s in a reasonable state for review.  You’re wasting everyone’s time if you’re opening pull requests for things that aren’t ready for review.

**Comment by mamehaze** [comment]
then this progress is dead too

**Comment by modusc896d352** [comment]
congrats cuavas, you made haze mad

**Comment by cuavas** [comment]
> Can we leave this open, please, so someone else can continue it in the futre.

It’s not worth it – it isn’t a good base to continue from.  I’ve already written a fully functional C33 Core disassembler.

**Comment by mamehaze** [comment]
it would have been a fine base to continue from if you hadn't nuked it.

I'd gone for a nice 'clean room' approach, whereby I'd written down the basic ops, their params, and the masks needed, and coded that into what you have here, giving an airgap between the code and the manual.  This was the first step of my process, which would have ultimately used the official syntax once I'd learned how things worked.

YOU just wanted step 4 before you had step 2, no appreciation of the process of your developers, or the order in which they wish to develop things.

Apparently it's just fine for YOU to do incremental updates to YOUR plan, just not anybody else.

Would have ended up at the same destination in the end, but you're just insufferable, fail to understand the working process others are going through, accuse others of wasting our time (when in fact, you're wasting theirs) and you're just all round awful for the project.  Do you understand how rude that accusation was? No.

I'm even having to deal with (another) case of you slamming the door on somebody who was just trying to learn, and made a mistake (used an issue rather than a PR)  Guess what, you wasted their time too, and rather than be helpful, just ensured they're not going to come back and commit anything.

The number of times I've been asked to make PRs for other people so that they don't have to deal with you is adding up, but in cases like the above, it doesn't even look like they're going to ask me, instead simply not bother.

I'm sure you'll just delete this / lock anyway though.

**Comment by mamehaze** [comment]
it would have been a fine base to continue from if you hadn't nuked it by making it very clear I couldn't follow my plan, and had to deviate from it for your sake.

I'd gone for a nice 'clean room' approach, whereby I'd written down the basic ops, their params, and the masks needed, and coded that into what you have here, giving an airgap between the code and the manual. This was the first step of my process, which would have ultimately used the official syntax once I'd learned how things worked.

YOU just wanted step 4 before you had step 2, no appreciation of the process of your developers, or the order in which they wish to develop things.

I appreciate actual bugs being pointed out (the infinite loop) because that was a simple oversight. I'd chosen to not stage the other couple of days of work (the execution core) as it wasn't ready yet, but didn't realise it had taken out the icounter bit too. I do not appreciate you dictating design or process when it comes to figuring things out and the overall learning process.

Apparently it's just fine for YOU to do incremental updates to YOUR plan, just not anybody else (unless their process matches what yours would be)

It would have ended up at the same destination in the end, but you're just insufferable, fail to understand the working process others are going through, accuse others of wasting our time (when in fact, you're wasting theirs) and you're just all round awful for the project. Do you understand how rude that accusation was? No.

Even here you're being incredibly rude with "it isn’t a good base to continue from" It's just not a base YOU wanted to continue from because it didn't match YOUR plan. It was a perfectly good base to continue to as far as my own plan went, and as I was the one developing it, maybe that's something you should have actually considered.

I'll respond to this kind of rudeness with rudeness in equal measure.

I'm even having to deal with (another) case of you slamming the door on somebody who was just trying to learn, and made a mistake (used an issue rather than a PR) Guess what, you wasted their time too, and rather than be helpful, just ensured they're not going to come back and commit anything.

The number of times I've been asked to make PRs for other people so that they don't have to deal with you is adding up, but in cases like the above, it doesn't even look like they're going to ask me, instead simply not bother.

I'm sure you'll just delete this / lock anyway though.

**Comment by cuavas** [comment]
> it would have been a fine base to continue from if you hadn't nuked it.

No it wasn’t.  It had far too much copy/paste for different instructions, so trying to maintain it would’ve required a huge amount of effort for any kind of change.

Also trying to show synthetics for long branches and extended addressing modes would have required far too much copy/paste with that code structure.

You can call it “personal style” if you want, but it’s a style that massively increases maintenance overhead and makes the code more difficult to debug.

> I'd gone for a nice 'clean room' approach, whereby I'd written down the basic ops, their params, and the masks needed, and coded that into what you have here, giving an airgap between the code and the manual. This was the first step of my process, which would have ultimately used the official syntax once I'd learned how things worked.

If you were going to do that, the time to do it is *before* opening the pull request.  You don’t open a pull request with something in a state that isn’t ready for merging.

> YOU just wanted step 4 before you had step 2, no appreciation of the process of your developers, or the order in which they wish to develop things.

Developers are quite capable of getting those steps done on a branch before opening a pull request or pushing to a live repository.  There’s no reason you need to have an open pull request for something that isn’t ready.  Opening a pull request is saying it’s ready for review, and if you do that when something isn’t, it wastes people’s time reviewing it.

You can develop on a branch of your own, push to your own repository on GitHub to get CI artefacts, etc. without an open pull request.

> Apparently it's just fine for YOU to do incremental updates to YOUR plan, just not anybody else (unless their process matches what yours would be)

My first commit of a C33 disassembler produced proper syntax for all C33 ADV Core instructions (a superset of the C33 STD Core and C33 EP Core).  It was usable from unidasm.  It did useful things immediately, following MAME conventions.  The only things I’ve added since then are reconstruction of additional assembler synthetics to make the output more readable.

> It would have ended up at the same destination in the end, but you're just insufferable, fail to understand the working process others are going through, accuse others of wasting your time (when in fact, you're wasting theirs) and you're just all round awful for the project. Do you understand how rude that accusation was? No.

Opening a pull request for something that isn’t ready for review *is* wasting people’s time, and is incredibly rude.  Do you understand that?

> Even here you're being incredibly rude with "it isn’t a good base to continue from" It's just not a base YOU wanted to continue from because it didn't match YOUR plan. It was a perfectly good base to continue to as far as my own plan went, and as I was the one developing it, maybe that's something you should have actually considered.

I have not been rude a single time.  I have not insulted you.  The trouble is, you call any feeback that isn’t praise “rude”.  On the other hand, your passive-aggressive commit messages on recent pull requests *are* rude.

> I'm sure you'll just delete this / lock anyway though.

Are you projecting?  How many times have I deleted your comments?

**Comment by mamehaze** [comment]
> Opening a pull request for something that isn’t ready for review is wasting people’s time, and is incredibly rude. Do you understand that?

It was ready for review, I wanted feedback on the BUGS, not demanding a different design.

Again, it would have evolved into something much closer to what you have now.  That was my final goal.  That wasn't your work to do, or dictate.  You just wanted steps in a different order, at a different rate.

>I have not been rude a single time. I have not insulted you. The trouble is, you call any feeback that isn’t praise “rude”. On the other hand, your passive-aggressive commit messages on recent pull requests are rude.

You're rude all the time, maybe you just can't see it.  I've had multiple complaints about you being rude.

**Comment by cuavas** [comment]
> It was ready for review, I wanted feedback on the BUGS, not demanding a different design.

Displaying something that looks nothing like the official assembly language for the target processor, doesn’t display calculated relative branch target addresses, doesn’t show real register names, doesn’t use the correct format for integer literals, etc. is a “bug” in that it doesn’t meet the requirements for a MAME disassembler.

> Again, it would have evolved into something much closer to what you have now. That was my final goal. That wasn't your work to do, or dictate. You just wanted steps in a different order, at a different rate.

I’m just asking that you have something in line with MAME’s standards before opening a pull request.

> I've exhausted my niceness for you, so yes, enjoy passive aggressive commit messages, because I have no fuse left when it comes to dealing with you. This is me trying to be as nice as possible at this point, but I'm utterly done with you and your methods.

What niceness?  You’ve been attacking and insulting me for over twenty years now.  You just claim not to remember.  I haven’t forgotten the way you’ve attacked and insulted me on the MAME World and Bannister forums for over two decades.  I’ve refrained from responding in kind.

>  You don't need to copy+paste the same complaint 50 times for example.

Yet when I *don’t* note every instance, you only address the instances I *do* mark, and you mark the feedback resolved.

>  For the other PR, I can't srcclean it, it's a webedit, but your comments come across like a cold dismissal, a rejection, and just because I used the web-editor and therefore can't, it won't be merged.

You can save the file, srcclean it, and upload it (or paste the contents) in the web editor.  You don’t need to do a full checkout.

> There have been constructive talks elsewhere about maybe having an unstable branch, with no big delays to acceptance (as long as the build doesn't break) with you having your own stable branch, where you can pull from unstable, do whatever cleanups you desire, then form that into release builds. You're treating all of MAME like it needs to be some "can't fail" industry standard thing, when really it's an "always in progress" incremental project, which we like to highlight with the 0.xxx versioning.

The trouble with that idea is that having a branch where stuff is accepted before it’s “ready” is that you’d get stuff that the author isn’t prepared to clean up, and then never makes it to mainline.  Then the “unstable” branch would diverge from mainline to the point where it isn’t practical to cherry-pick things any more, and merging changes from mainline into the “unstable” branch would become an inordinate amount of work.  The only way to avoid that would be to revert stuff if it doesn’t get cleaned up in some reasonable amount of time, but that wouldn’t make people happy, either.

**Comment by mamehaze** [comment]
> What niceness? You’ve been attacking and insulting me for over twenty years now. You just claim not to remember. I haven’t forgotten the way you’ve attacked and insulted me on the MAME World and Bannister forums for over two decades. I’ve refrained from responding in kind.

The only time I remember attacking you is in response to direct attacks to from you and RB over UME.

I do remember some incredibly creepy posts from you on MW where you were objectifying women etc. but if *that* is what you're talking about, I stand by it.  Those posts got Mameworld blocked at a place of work.  (being called into an office to justify why a site I had been browsing was glorifying upskirting was certainly the conversation to have)

**Comment by mamehaze** [comment]
> The trouble with that idea is that having a branch where stuff is accepted before it’s “ready” is that you’d get stuff that the author isn’t prepared to clean up, and then never makes it to mainline. Then the “unstable” branch would diverge from mainline to the point where it isn’t practical to cherry-pick things any more, and merging changes from mainline into the “unstable” branch would become an inordinate amount of work. The only way to avoid that would be to revert stuff if it doesn’t get cleaned up in some reasonable amount of time, but that wouldn’t make people happy, either.

The unstable branches on quite a few projects seem to be doing better than the stable ones however.  Often because stable has set the standards too high.  It would be the role of the person who expects better standards to tidy things up to meet them, it's a legitimate role, for somebody who is kinda obsessed with that kind of thing.  I don't know anybody more obsessed that you, for better or worse.

**Comment by cuavas** [comment]
> It would be the role of the person who expects better standards to tidy things up to meet them, it's a legitimate role, for somebody who is kinda obsessed with that kind of thing.

There are no projects, professional or open source, that work like this.  “One person cleans up the mess other people make” doesn’t work.

**Comment by mamehaze** [comment]
> Displaying something that looks nothing like the official assembly language for the target processor, doesn’t display calculated relative branch target addresses, doesn’t show real register names, doesn’t use the correct format for integer literals, etc. is a “bug” in that it doesn’t meet the requirements for a MAME disassembler.

Again, this was step 1, the first milestone, what my working process had produced.

I was still trying to fully understand exactly how the branch calculations work (as I was working on the core) before moving that logic into the disassembler.  I explicitly named each type, so that when replacing them later, it would be easier.

There was a clear method, I don't consider it a bug, it was incremental progress with a clear goal.

Yes, your process is different, but you weren't the one working on it.

It was still the milestone I wanted to have included before moving on to other parts, then coming back to it.

2nd PR would have been semi-functional execution core
3rd PR would have been moving towards better syntax
Etc.

**Comment by mamehaze** [comment]
> There are no projects, professional or open source, that work like this. “One person cleans up the mess other people make” doesn’t work.

most places I've worked employ such practices.

we don't call it "mess" we call it helping the other person out, leading.  having the team work together as a whole, allowing each person to focus on their individual skills.

most of the time it isn't a mess either, that's just you being rude again.

**Comment by mamehaze** [comment]
Also, no, I'm not claiming to be perfect.  I've made mistakes in the past.

Some of those mistakes I've made in the past are ones I'm seeing you make now.

I've been too harsh on people in the past when it comes to contributions, and expectations, while at the same time trying to put everything into helping them.  We do all let ourselves down from time to time.

It's a learning process, and if anything it's painful seeing you make the same mistakes, because they do hurt the project.

Aaron was definitely a much better co-ordinator in terms of dealing with people, as was Micko, but I had my issues with them.  

The shortsightness of ignoring the console side of thing (essential to development) put us entirely on the back foot against the likes of RetroArch, and pretending MAME was an 'accurate emulator' to excuse the fact it was so slow was a lie that people had really started to see through when we weren't even doing basic things such as waitstates, that others had done for over a decade, but instead putting our head in the sand.

In this case, the shortsightness is gone, but making the project this unapproachable to those without skills the same grade as yours is just as much of an issue, so yes, I'm going to raise it.

**Comment by mamehaze** [comment]
> You can save the file, srcclean it, and upload it (or paste the contents) in the web editor. You don’t need to do a full checkout.

I can't even see an option to re-edit, but the interface of GItHub is so messy maybe I missed it.  My version control software certainly doesn't give me the option of a single file checkout, and I'm not learning the commandline just for this, as I consider what I am doing to be the limit of my understanding of Git.

Also last time I just pasted my local copy over the edit window it said I'd changed every line (you've had others having this issue)

Maybe just quicker to merge it, and let the release srcclean take care of it.  The release srcclean is there for a reason.

This is very, very minor, and not even worth this discussion.

**Comment by simzy39** [comment]
I like the way Linux does it with the 'Linux-next' branch, and then once tried and proven the patches get merged into Linus's main branch.

**Pull request closed without merging**